[issue scrubbing] Fix imports in test_memory_planning.py to work with pytest

masnesral opened the pull request:
Stack from [ghstack](https://github.com/ezyang/ghstack) (oldest at bottom):
* __->__ #130275

Summary: I actually don't grok why this pattern works; I guess pytest expects a different import syntax for these relative imports?? But this pattern is used in many other tests here (notably `test_aot_inductor.py`), so it must be right ;)

Test Plan:
Ran both ways:
* `python test/inductor/test_memory_planning.py`
* `pytest test/inductor/test_memory_planning.py`

cc @voznesenskym @penguinwu @EikanWang @jgong5 @Guobing-Chen @XiaobingSuper @zhuhaozhe @blzheng @wenzhe-nrv @jiayisunx @peterbell10 @ipiszy @yf225 @chenyang78 @kadeng @muchulee8 @ColinPeppler @amjames @desertfire @chauhang

masnesral committed:
[issue scrubbing] Fix imports in test_memory_planning.py to work with pytest

Summary: I actually don't grok why this pattern works; I guess pytest expects a different import syntax for these relative imports?? But this pattern is used in many other tests here (notably `test_aot_inductor.py`), so it might be right ;)

Test Plan:
Ran both ways:
* `python test/inductor/test_memory_planning.py`
* `pytest test/inductor/test_memory_planning.py`

[ghstack-poisoned]

masnesral commented:
To fix https://github.com/pytorch/pytorch/issues/117078

masnesral committed:
lint on "[issue scrubbing] Fix imports in test_memory_planning.py to work with pytest"


Summary: I actually don't grok why this pattern works; I guess pytest expects a different import syntax for these relative imports?? But this pattern is used in many other tests here (notably `test_aot_inductor.py`), so it might be right ;)

Test Plan:
Ran both ways:
* `python test/inductor/test_memory_planning.py`
* `pytest test/inductor/test_memory_planning.py`

cc voznesenskym penguinwu EikanWang jgong5 Guobing-Chen XiaobingSuper zhuhaozhe blzheng wenzhe-nrv jiayisunx peterbell10 ipiszy yf225 chenyang78 kadeng muchulee8 ColinPeppler amjames desertfire chauhang

[ghstack-poisoned]

zou3519 commented:
> Summary: I actually don't grok why this pattern works; I guess pytest expects a different import syntax for these relative imports??

I have had the same exact question before and I'm not sure why

zou3519 commented on test/inductor/test_memory_planning.py:
```diff
@@ -24,6 +24,17 @@
+    except ImportError:
+        from test_aot_inductor import AOTIRunnerUtil
+except (unittest.SkipTest, ImportError) as e:
+    sys.stderr.write(f"{type(e)}: {e}\n")
+    if __name__ == "__main__":
+        sys.exit(0)
```
What's the except for? If "this file should run even without AOTInductor" then we can probably put the try-except into `test_abi_compatible` ?

masnesral committed:
comments on "[issue scrubbing] Fix imports in test_memory_planning.py to work with pytest"


Summary: I actually don't grok why this pattern works; I guess pytest expects a different import syntax for these relative imports?? But this pattern is used in many other tests here (notably `test_aot_inductor.py`), so it must be right ;)

Test Plan:
Ran both ways:
* `python test/inductor/test_memory_planning.py`
* `pytest test/inductor/test_memory_planning.py`

[ghstack-poisoned]

masnesral commented:
@pytorchbot merge
